rinja_parser/src/lib.rs

```diff
@@ -134,19 +134,70 @@ impl<'a> Ast<'a> {
 /// in the code generation.
 pub struct WithSpan<'a, T> {
     inner: T,
-    span: &'a str,
+    span: Span<'a>,
+}
```

Comment:
To ensure the size doesn't grow up, might be worth adding a compile-time `size_of` check of `WithSpan` and of `Span`.

Reply:
I don't think it matters too much, as the PR fixes another problem, too:

If the input span is shrunk at the end of the input, e.g. `let i = i.trim()`, then the error messages will be broken afterward, because only the lengths are compared. This PR fixes that, too. This was actually my original intention with, because I had to refactor some code a few times in order not to break the error messages. That the Span is smaller afterwards is just a happy co-incidence. :)